doc/source/internals.rst

```diff
@@ -94,8 +94,79 @@ not check (or care) whether the levels themselves are sorted. Fortunately, the
 constructors ``from_tuples`` and ``from_arrays`` ensure that this is true, but
 if you compute the levels and labels yourself, please be careful.
 
+.. _ref-composition-pandas:
 
-.. _:
+Define Original Data Structures using pandas
+--------------------------------------------
+
+.. warning:: If you simply want to add some functionalities to ``pandas``, the easiest
+   way is monkey-patching. See :ref:`Adding Features to your pandas Installation <ref-monkey-patching>`.
+
+This section describes how to define your original data structure which extends ``pandas`` functionalities using `composition <http://en.wikipedia.org/wiki/Composition_over_inheritance>`_.
+
+Below example shows an original class which is mostly compatible with ``Series``.
+The class have a ``_series`` property to hold standard ``Series`` (composition), and defining ``__getattribute__`` to delegate all the undefined properties / methods to it.
+
+.. code-block:: python
+
+   class CompositedSeries(object):
+
+       def __init__(self, arr, *args, **kwargs):
+           self._series = Series(arr, *args, **kwargs)
+
+       def __getattribute__(self, key):
```

Comment:
Is it not possible to use `__getattr__` instead? Usually that is preferable because it's called only as a fallback (unlike `__getattribute__`).